src/app/shared/ga4gh-files/ga4gh-files.query.ts

```diff
@@ -27,14 +27,23 @@ import { GA4GHFilesState, GA4GHFilesStore } from './ga4gh-files.store';
   providedIn: 'root'
 })
 export class GA4GHFilesQuery extends QueryEntity<GA4GHFilesState, GA4GHFiles> {
+  /**
+   * Returns an Observable array of ToolFile, unless the store is an error state, in which case it returns
+   * a null. The store can be in an error state, for example, if there was an error in the Ajax request.
+   * @param descriptorType
+   * @param fileTypes
+   */
   getToolFiles(descriptorType: ToolDescriptor.TypeEnum, fileTypes: Array<ToolFile.FileTypeEnum>): Observable<Array<ToolFile>> {
     if (descriptorType) {
       let toolFiles$: Observable<Array<ToolFile>>;
-      toolFiles$ = this.selectEntity(descriptorType).pipe(
-        map((gA4GHFile: GA4GHFiles) => gA4GHFile ? gA4GHFile.toolFiles : [])
-      );
+      toolFiles$ = this.selectError().pipe(switchMap(
+        error => {
+          return error != null ? of(null) : this.selectEntity(descriptorType).pipe(
```

Comment:
use `observableOf` instead of `of`, then remove `of` from imports

Reply:
Done with a new commit